shark_turbine/kernel/wave/constraints.py

```diff
@@ -101,3 +103,31 @@ def get_grid_shape(wg_constraints: list[WorkgroupConstraint]) -> list[IndexExpr]
         constraint.dim // constraint.tile_size for constraint in wg_constraints
     ]
     return grid
+
+
+@dataclass
+class TilingConstraint(Constraint):
+    """
+    A constraint of the form `tkw.TilingConstraint(K, BLOCK_K)` specifies
+    that we want to tile the K dimension with a tile size of BLOCK_K. This
+    adds an index constraint to the K-th dimension of a tensor of the form
+    BLOCK_K * i, where i is the induction variable associated with the
+    loop around dimension K.
+    """
+
+    dim: IndexExpr
+    tile_size: IndexExpr
+    induction_var: Optional[IndexExpr] = None
+
+    def iterations(self) -> IndexExpr:
+        """
+        Returns an expression for the number of iterations in the loop.
+        """
+        return ceiling(self.dim / self.tile_size)
```

Comment:
Would we handle the non-aligned-ness/padding of it somewhere else?

Reply:
Yes, this just gives the number of iterations.